### PR TITLE
Adding Azure DevOps data connector and onboarding scripts

### PR DESCRIPTION
**Pull request by v-mchatla**
Azure DevOps Auditing Repackaging changes

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - [TestEvidenceDoc](https://microsoftapc.sharepoint.com/:w:/r/teams/EcosystemCATTeam/_layouts/15/Doc.aspx?sourcedoc=%7BFBF83981-BE32-491B-A367-BF67EB8D8803%7D&file=Sentinel%20Release%20Checklist%20-V2%20solutions-%20Azure%20DevOps%20Auditing.docx&action=default&mobileredirect=true&share=IQGBOfj7Mr4bSaNnv2frjYgDAUx8W0GX8VkNodbfUJnD2kQ)

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-spadarthi** [comment]
@v-mchatla : Please review the PR and provide your feedback. Thanks

**Comment by v-sabiraj** [comment]
@v-mchatla, please update the file extension for "Set-AzureDevOpsAuditing.ps1.txt" as it should be just "Set-AzureDevOpsAuditing.ps1", same with "readme.md.txt", it should just be "readme.md", thanks.

File: Solutions/AzureDevOpsAuditing/Data Connectors/_createUiDefinition.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/0.1.2-preview/CreateUIDefinition.MultiVm.json#",
+    "handler": "Microsoft.Azure.CreateUIDef",
+    "version": "0.1.2-preview",
+    "parameters": {
+        "config": {
+            "isWizard": true,
+            "basics": {
+                "description": "<img src=\"https://raw.githubusercontent.com/SecureHats/Azure-Sentinel/master/DataConnectors/AzureDevOpsAuditing/media/banner-small.svg\">\n\n**Azure DevOps Auditing - Connector** \n\nThe Azure DevOps audit log connector provides the capability to ingest Azure DevOps auditing logs into Microsoft Sentinel. \nBy connecting Azure DevOps audit logs into Microsoft Sentinel, you can view this data in workbooks, use it to create custom alerts, and improve your investigation process. \n\n**Information:** \nIn order to configure Azure DevOps the deployment uses an ARM deployment script which requires a user assigned identity. \nYou can remove this managed identity after deployment, if desired.\n\n[Learn more about Azure Sentinel](https://aka.ms/azuresentinel)",
```
Comment (v-sabiraj) [inline Solutions/AzureDevOpsAuditing/Data Connectors/_createUiDefinition.json:9]: @v-mchatla, please update the svg file, thanks.

File: Solutions/AzureDevOpsAuditing/Data Connectors/connector_AzureDevOpsAuditing.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                        "label": "Run the following command to configure Azure DevOps Streams:",
+                        "value": "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/AzureDevOpsAuditing/Data%20Connectors/scripts/Set-AzureDevOpsAuditing.ps1' -OutFile 'Set-AzureDevOpsAuditing.ps1' && .\/Set-AzureDevOpsAuditing.ps1 -workspaceId {0} -workspaceKey {1} -organization {organization} -personalaccesstoken {pattoken}"
+                    },
+                    "type": "CopyableLabel"
+                }
+            ]
+        },
+        {
+            "title": "Option 2 - Azure Resource Manager (ARM) Template",
+            "description": "Use this method for automated configuration of the data connector using an ARM Template.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinel-AzureDevOpsAuditing-azuredeploy)",
```
[the remark this answers is not in the document]
Comment (v-mchatla) [reply Solutions/AzureDevOpsAuditing/Data Connectors/connector_AzureDevOpsAuditing.json:92]: links will work once merged to master

File: Solutions/AzureDevOpsAuditing/Data Connectors/readme.md.txt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+## Configuring the Azure DevOps Audit Streams
+
+There are several options to configure the Audit Stream in Azure DevOps.
+The possible options can also be found in the Data Connector after this has been deployed to Microsoft Sentinel.
+
+### Option 1 - Azure Resource Manager (ARM) Template
+
+Use this method for automated configuration of the data connector using an ARM Template.
+
+[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FAzureDevOpsAuditing%2FData%20Connectors%2Fazuredeploy.json/createUIDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FAzureDevOpsAuditing%2FData%20Connectors%2F_createUiDefinition.json)
```
[the remark this answers is not in the document]
Comment (v-mchatla) [reply Solutions/AzureDevOpsAuditing/Data Connectors/readme.md.txt]: Links will work once merged to master

File: Solutions/AzureDevOpsAuditing/Data Connectors/readme.md.txt [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Azure DevOps Microsoft Sentinel Connector
+
+Author: Rogier Dijkman
+
+The Azure DevOps Auditing Sentinel connector provides the capability to ingest Azure DevOps events in Microsoft Sentinel.
+It helps you gain visibility into what is happening in your environment, such as who is connected, which applications are installed and running, and much more.
+
+[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FAzureDevOpsAuditing%2FData%20Connectors%2FmainTemplate.json/createUIDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FAzureDevOpsAuditing%2FData%20Connectors%2FuiMainTemplate.json)
```
[the remark this answers is not in the document]
Comment (v-mchatla) [reply Solutions/AzureDevOpsAuditing/Data Connectors/readme.md.txt]: Links will work once merged to master

**Comment by v-mchatla** [comment]
Hi @v-sabiraj ,
I will update the author details.
Thanks

**Comment by v-mchatla** [comment]
Hi @v-sabiraj,
I will address the remaining comments today.
Thanks

**Comment by v-sabiraj** [comment]
@v-mchatla, please check, thanks.

File: Solutions/AzureDevOpsAuditing/Data Connectors/connector_AzureDevOpsAuditing.json [hunk 5]
```diff
@@ -0,0 +1,113 @@
+{
+    "id": "AzureDevOpsAuditing",
+    "title": "Azure DevOps Auditing",
+    "publisher": "SecureHats",
```
[the remark this answers is not in the document]
Comment (v-mchatla) [reply Solutions/AzureDevOpsAuditing/Data Connectors/connector_AzureDevOpsAuditing.json]: fixed.

**Comment by v-mchatla** [comment]
Hi @v-sabiraj,
Waiting on confirmation from Prateek.
Thanks

**Comment by v-sabiraj** [comment]
@v-mchatla, please update branch from master to fix the arm-ttk validations.

File: Solutions/AzureDevOpsAuditing/Data Connectors/mainTemplate.json [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      "id": "[variables('_connector1-source')]",
+      "name": "[concat(parameters('workspace'), '/Microsoft.SecurityInsights/', parameters('connector1-name'))]",
+      "apiVersion": "2021-03-01-preview",
+      "type": "Microsoft.OperationalInsights/workspaces/providers/dataConnectors",
+      "kind": "GenericUI",
+      "properties": {
+        "connectorUiConfig": {
+          "title": " Azure DevOps Auditing",
+          "publisher": "SecureHats",
+          "descriptionMarkdown": "The Azure DevOps Auditing data connector provides the capability to ingest Azure DevOps events with Azure Sentinel. It helps you gain visibility into what is happening in your environment, such as who is connected, which applications are installed and running, and much more.",
```
Comment (v-sabiraj) [inline Solutions/AzureDevOpsAuditing/Data Connectors/mainTemplate.json:37]: @v-mchatla, please change text from Azure Sentinel to Microsoft Sentinel, thanks.

File: Solutions/AzureDevOpsAuditing/Data Connectors/_createUiDefinition.json [hunk 7]
```diff
@@ -0,0 +1,86 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/0.1.2-preview/CreateUIDefinition.MultiVm.json#",
+    "handler": "Microsoft.Azure.CreateUIDef",
+    "version": "0.1.2-preview",
+    "parameters": {
+        "config": {
+            "isWizard": true,
+            "basics": {
+                "description": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Azure_Sentinel.svg\" width=\"75px\" height=\"75px\">\n\n**Azure DevOps Auditing - Connector** \n\nThe Azure DevOps audit log connector provides the capability to ingest Azure DevOps auditing logs into Microsoft Sentinel. \nBy connecting Azure DevOps audit logs into Microsoft Sentinel, you can view this data in workbooks, use it to create custom alerts, and improve your investigation process. \n\n**Information:** \nIn order to configure Azure DevOps the deployment uses an ARM deployment script which requires a user assigned identity. \nYou can remove this managed identity after deployment, if desired.\n\n[Learn more about Azure Sentinel](https://aka.ms/azuresentinel)",
```
Comment (v-sabiraj) [inline Solutions/AzureDevOpsAuditing/Data Connectors/_createUiDefinition.json:9]: @v-mchatla, please change text from Azure Sentinel to Microsoft Sentinel, thanks.

**Comment by v-atulyadav** [comment]
Looking into this

**Comment by v-sabiraj** [comment]
@v-mchatla, please update the branch from master.

**Comment by v-mchatla** [comment]
Hi @v-sabiraj,
I will fix the validation errors today.
Thanks

**Comment by v-sabiraj** [comment]
@v-mchatla, waiting on merge confirmation.

**Comment by v-mchatla** [comment]
Hi @v-sabiraj,
I have requested @rushriva for some help in configuring the connector. Once we got approval will go ahead with the merge.
Thanks

**Comment by v-prasadboke** [comment]
Hello @v-mchatla please clear the checks

**Comment by v-mchatla** [comment]
Hi @rushriva,
Can you please provide update on how to proceed further.
Thanks

**Comment by v-mchatla** [comment]
Hi @rushriva,
Can you please provide update on how to proceed further.
Thanks

**Comment by v-mchatla** [comment]
Hi,
I will go through the data connector instruction steps once and try to configure the connector.
Thanks

**Comment by v-mchatla** [comment]
Hi @v-rucdu,
Can you please try deploying the connector once and let me know the update.
Thanks

**Comment by v-mchatla** [comment]
Hi @v-rucdu,
Can you please try deploying the connector once and let me know the update.
Thanks

**Comment by v-mchatla** [comment]
Hi @v-rucdu,
Can you please try deploying the connector once and let me know the update.
Thanks

**Comment by v-sabiraj** [comment]
@v-mchatla, any update on this?

**Comment by v-sabiraj** [comment]
Waiting for update on this, thanks.

**Comment by v-mchatla** [comment]
Hi @v-rucdu,
Can you please try deploying the connector once and let me know the update.
Thanks

**Comment by v-mchatla** [comment]
Hi,
Yesterday Ruchita had deployed the solution but we need do deploy the connector and see the functionality.
Thanks

**Comment by v-atulyadav** [comment]
Waiting for update on this, thanks.

**Comment by v-mchatla** [comment]
Hi,
Requested help from Ruchita for Dataconnector testing.
Thanks

**Comment by v-mchatla** [comment]
Hi,
Requested help from Ruchita for Dataconnector testing.
Thanks

**Comment by v-prasadboke** [comment]
Hello @v-rucdu please reply on the following

**Comment by v-atulyadav** [comment]
Hi,
Requested help from Ruchita for Dataconnector testing.
Thanks

**Comment by v-atulyadav** [comment]
Waiting for update on this, thanks.

**Comment by v-atulyadav** [comment]
Hi,
Requested help from Ruchita for Dataconnector testing.
Thanks

**Comment by v-atulyadav** [comment]
Hi,
Requested help from Ruchita for Dataconnector testing.
Thanks

**Comment by v-atulyadav** [comment]
closing for now

**Pull request closed without merging**